test/unit/lib/entities/context/RelayQuoteContext.test.ts

```diff
@@ -0,0 +1,162 @@
+    it('returns expected dependencies when output is weth', () => {
+      const context = new RelayQuoteContext(logger, QUOTE_REQUEST_RELAY, makeProviders());
+      const deps = context.dependencies();
+      expect(deps.length).toEqual(2);
+      // first is base
+      expect(deps[0]).toEqual(QUOTE_REQUEST_RELAY);
+      // second is classic
+      expect(deps[1].info).toEqual(QUOTE_REQUEST_RELAY.info);
+      expect(deps[1].routingType).toEqual(RoutingType.CLASSIC);
+    });
+
+    it('returns expected dependencies when output is not weth', () => {
+      const request = makeRelayRequest({
+        tokenOut: '0x1111111111111111111111111111111111111111',
+      });
+      const context = new RelayQuoteContext(logger, request, makeProviders());
+      const deps = context.dependencies();
+      expect(deps.length).toEqual(2);
+      // first is base
+      expect(deps[0]).toEqual(request);
+      // second is classic
+      expect(deps[1].info).toEqual(request.info);
+      expect(deps[1].routingType).toEqual(RoutingType.CLASSIC);
+    });
+  });
+
+  describe('resolve', () => {
+    it('returns null if no dependencies given', async () => {
+      const context = new RelayQuoteContext(logger, QUOTE_REQUEST_RELAY, makeProviders());
+      expect(await context.resolve({})).toEqual(null);
+    });
+
+    it('returns null if quote key is not set properly', async () => {
+      const context = new RelayQuoteContext(logger, QUOTE_REQUEST_RELAY, makeProviders());
+      expect(
+        await context.resolve({
+          wrong: DL_QUOTE_EXACT_IN_BETTER,
+        })
+      ).toBeNull();
+    });
+
+    it('returns main quote if others are null', async () => {
+      const context = new RelayQuoteContext(logger, QUOTE_REQUEST_RELAY, makeProviders());
+
+      const relayQuote = createRelayQuote({ amountOut: AMOUNT }, 'EXACT_INPUT');
+      const quote = await context.resolve({
+        [QUOTE_REQUEST_RELAY.key()]: relayQuote,
+      });
+      expect(quote).toMatchObject(relayQuote);
+    });
+
+    it('reconstructs quote from dependencies if main quote is null', async () => {
+      const context = new RelayQuoteContext(logger, QUOTE_REQUEST_RELAY, makeProviders());
+
+      const classicQuote = createClassicQuote(
+        { quote: '10000000000', quoteGasAdjusted: '9999000000', gasUseEstimateGasToken: '1' },
+        { type: 'EXACT_INPUT' }
+      );
+      context.dependencies();
+
+      const quote = await context.resolve({
+        [context.classicKey]: classicQuote,
+      });
+      expect(quote?.routingType).toEqual(RoutingType.RELAY);
+    });
+
+    it('returns null if quotes have 0 amountOut', async () => {
+      const context = new RelayQuoteContext(logger, QUOTE_REQUEST_RELAY, makeProviders());
+
+      const relayQuote = createRelayQuote({ amountOut: '0' }, 'EXACT_INPUT');
+      expect(
+        await context.resolve({
+          [QUOTE_REQUEST_RELAY.key()]: relayQuote,
+        })
+      ).toBe(null);
+    });
+
+    // TODO:
```

Comment:
can be removed?

Reply:
yeah whats the plan with these xits?

Reply:
they were copied over from dutch ones, will remove for now